packages/remix-react/components.tsx

```diff
@@ -1210,129 +1191,36 @@ export function useFetcher<TData = any>(): FetcherWithComponents<
-      // Devs should *not* be doing this but we don't want to break their
-      // current app if they are
-      type = value;
-    },
-    // These settings should make this behave like a normal object `type` field
-    configurable: true,
-    enumerable: true,
-  });
-
-  let submission: Fetcher["submission"] = fetcher.submission;
-  Object.defineProperty(fetcher, "submission", {
-    get() {
-      warnOnce(
-        false,
-        "⚠️ DEPRECATED: The `useFetcher().submission` field has been deprecated and " +
-          "will be removed in Remix v2.  The submission fields now live directly " +
-          "on the fetcher (`fetcher.formData`).\n\n" +
-          "See https://remix.run/docs/hooks/use-fetcher for more information."
-      );
-      return submission;
-    },
-    set(value: Fetcher["submission"]) {
-      // Devs should *not* be doing this but we don't want to break their
-      // current app if they are
-      submission = value;
-    },
-    // These settings should make this behave like a normal object `type` field
-    configurable: true,
-    enumerable: true,
-  });
-}
-
 function convertRouterFetcherToRemixFetcher(
   fetcherRR: Omit<ReturnType<typeof useFetcherRR>, "load" | "submit" | "Form">
 ): Fetcher {
-  let { state, formMethod, formAction, formEncType, formData, data } =
+  let { state, data, formMethod, formAction, formData, formEncType } =
     fetcherRR;
 
-  let isActionSubmission =
-    formMethod != null &&
-    ["POST", "PUT", "PATCH", "DELETE"].includes(formMethod.toUpperCase());
-
-  if (state === "idle") {
-    if (fetcherRR[" _hasFetcherDoneAnything "] === true) {
-      let fetcher: FetcherStates["Done"] = {
-        state: "idle",
-        type: "done",
-        formMethod: undefined,
-        formAction: undefined,
-        formData: undefined,
-        formEncType: undefined,
-        submission: undefined,
-        data,
-      };
-      return fetcher;
-    } else {
-      let fetcher: FetcherStates["Idle"] = IDLE_FETCHER;
-      return fetcher;
-    }
-  }
-
-  if (
-    state === "submitting" &&
-    formMethod &&
-    formAction &&
-    formEncType &&
-    formData
-  ) {
-    if (isActionSubmission) {
-      // Actively submitting to an action
-      let fetcher: FetcherStates["SubmittingAction"] = {
+  if (state === "submitting") {
+    if (formMethod && formAction && formData && formEncType) {
+      let fetcher: FetcherStates["Submitting"] = {
         state,
-        type: "actionSubmission",
-        formMethod: formMethod.toUpperCase() as ActionSubmission["method"],
-        formAction: formAction,
-        formEncType: formEncType,
-        formData: formData,
-        submission: {
-          method: formMethod.toUpperCase() as ActionSubmission["method"],
-          action: formAction,
-          encType: formEncType,
-          formData: formData,
-          key: "",
-        },
         data,
+        formMethod: formMethod.toUpperCase() as FormMethod,
```

Comment:
Small back-compat layer kept to preserve the Remix uppercase `FormMethod` on `useFetcher`